ansible/roles/test/files/ptftests/py3/advanced-reboot.py

```diff
@@ -440,7 +440,13 @@ def generate_vlan_servers(self):
                 mac = self.VLAN_BASE_MAC_PATTERN.format(counter)
                 port = self.ports_per_vlan[vlan][i %
                                                  len(self.ports_per_vlan[vlan])]
-                addr = self.host_ip(prefix, i)
+                try:
+                    addr = self.host_ip(prefix, i)
+                except Exception as e:
+                    # If the number of hosts exceeds the number of available IPs in the subnet
+                    # we will get an exception. In this case, we will randomly assign an IP
```

Comment:
I had a change in #17236 that would add a note saying that the existing code would break if the number of hosts is more than half of the subnet size, after I made this discovery when going through the code. :)